File: src/main/java/wtf/gofancy/koremods/api/KoremodsLaunchPlugin.java
package wtf.gofancy.koremods.api;

import wtf.gofancy.koremods.prelaunch.KoremodsPrelaunch;

import java.nio.file.Path;
import java.util.Map;

public interface KoremodsLaunchPlugin {
    boolean shouldEnableSplashScreen();
    
    SplashScreen createSplashScreen(KoremodsPrelaunch prelaunch);
    
    void appendLogMessage(String message);
    
    void verifyScriptPacks(Map<String, Path> namespaceSources);
}

File: src/main/kotlin/wtf/gofancy/koremods/api/KoremodsLaunchPlugin.kt
package wtf.gofancy.koremods.api

import org.apache.logging.log4j.Level
import wtf.gofancy.koremods.prelaunch.KoremodsPrelaunch

interface KoremodsLaunchPlugin {
    fun shouldEnableSplashScreen(): Boolean
    
    fun createSplashScreen(prelaunch: KoremodsPrelaunch): SplashScreen
    
    fun appendLogMessage(level: Level, message: String)
}
